builtin/providers/google/resource_sql_database.go

```diff
@@ -67,6 +68,7 @@ func resourceSqlDatabaseCreate(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	err = sqladminOperationWait(config, op, "Insert Database")
+	googleMutexKV.Unlock(instanceMutexKey(project, instance_name))
```

Comment:
Does `defer` here actually result in the `Unlock` not being called until `resourceSqlDatabaseRead` is called? If so, then I think the unlock here is probably correct, _assuming_ the `resourceSqlDatabaseRead` is idempotent